nucleus/metrics/segmentation_metrics.py

```diff
@@ -306,41 +375,40 @@ def __init__(
                 (AND), forming a more selective `and` multiple field predicate.
                 Finally, the most outer list combines these filters as a disjunction (OR).
         """
-        assert (
-            0 <= iou_threshold <= 1
-        ), "IoU threshold must be between 0 and 1."
-        self.iou_threshold = iou_threshold
         super().__init__(
-            enforce_label_match,
-            confidence_threshold,
             annotation_filters=annotation_filters,
             prediction_filters=prediction_filters,
         )
-        self.metric = PolygonRecall(
-            self.enforce_label_match,
-            self.iou_threshold,
-            self.confidence_threshold,
-            self.annotation_filters,
-            self.prediction_filters,
+
+    def _metric_impl(
+        self,
+        annotation_img: np.ndarray,
+        prediction_img: np.ndarray,
+        annotation: SegmentationAnnotation,
+        prediction: SegmentationPrediction,
+    ) -> ScalarResult:
+
+        confusion = self._calculate_confusion_matrix(
+            annotation, annotation_img, prediction, prediction_img
         )
 
-    def call_poly_metric(
-        self, annotations: AnnotationList, predictions: PredictionList
-    ):
-        return self.metric(annotations, predictions)
+        with np.errstate(divide="ignore", invalid="ignore"):
```

Comment:
similar comment to above, for pickle's functional requirements, I think they're looking for instance-level recall/precision rather than pixel-level recall precision.  I judge this from trying to read through the coco-evaluation code they pointed us to (which is EXTREMELY hard to follow): https://github.com/facebookresearch/detectron2/blob/523c4023bc79719d2fc3fe0181311f88ad35a271/detectron2/evaluation/coco_evaluation.py#L222

So a). curious if you agree with me that the coco_evaluator seems to be doing instance level matching, and b). if that is indeed the case, I think we need to inject an extra instance matching step here to meet their requirements